src/database/queries/getData.js

```diff
@@ -24,6 +24,16 @@ const getSpecalize = (categoryId) => {
   };
   return connect.query(sql);
 };
+const getPropsalsbyValue = (specid, serchvalue) => {
+  const sql = {
+    text: 'select proposal.id, proposal.title, proposal.description,'
+    + ' users.firstname, users.lastname, users.photo_url '
+    + ' from proposal inner join users  on proposal.user_id = users.id'
+    + ` where proposal.specalization_id = $1 and lower(proposal.description) like '%${serchvalue}%' or lower(proposal.title) like '%${serchvalue}%'`,
+    values: [specid],
```

Comment:
@Fatmasiam To prevent sql injection, you should put searchValue into the values array too, instead of putting it inline in the query.
I think you are supposed to put the percent signs on the parameter inside the values array, so like this:
```
... where proposal.specalization_id = $1 and lower(proposal.description) like $2 or ...,
values: [specid, `%${searchvalue}%`],
```

Reply:
@tomduggan85  thanx a lot for your suggestion I have trying to prevent the injection but this is the right way .

Reply:
@tomduggan85 thx

Reply:
@Fatmasiam Cool, thanks for doing that!